distribution.pm

```diff
@@ -127,7 +129,8 @@ sub script_run ($self, $cmd, @args) {
         my $marker = "; echo $str-\$?-" . ($args{output} ? "Comment: $args{output}" : '');
         if (testapi::is_serial_terminal) {
             testapi::type_string($marker);
-            testapi::wait_serial($cmd . $marker, no_regex => 1, quiet => $args{quiet});
+            die 'Command was mistyped'
+              unless testapi::wait_serial($cmd . $marker, no_regex => 1, quiet => $args{quiet}, timeout => 5);
```

Comment:
the "5" is a magic number. I suggest to use an overridable constant with an explanatory name and defaulting to 5 if not set from the outside

Reply:
The point is that anyone else will wonder what the 5 means. So just by using a properly named constant you can avoid that confusion when reading the code. Likely no one ever would need to override the constant dynamically. This is only about code readability.

Reply:
Properly named constant such as `SHOULD_MATCH_INSTANTLY` or `DEFAULT_TIMEOUT_IS_WAY_TOO_LONG`? If you're concerned about code readability, then the real problem is Perl. The best I can do here is a code comment.

Reply:
if you say this should match instantly would "0" or "1" actually work?

Reply:
> If you're concerned about code readability, then the real problem is Perl.

Please no language bashing. It doesn't help anyone.
It's easy to write readable code in Perl.
When I'm reading code in a language I'm not very familiar with I also have problems with its readability.

Reply:
> if you say this should match instantly would "0" or "1" actually work?

"0" will not work for SSH consoles. "1" should work most of the time but it's not robust enough in case of network issues.

Reply:
```suggestion
              unless testapi::wait_serial($cmd . $marker, no_regex => 1, quiet => $args{quiet}, timeout => $args{wait_serial_timeout});
```

The reasoning here as far as I understand is a prime case for making it overridable. In unusual cases 1 is not enough but we really don't want the default of 90 seconds either. I'd suggest an argument if not a test setting (but I can't make it a nice looking suggesting here).

Reply:
In the usual cases, 1 second is more than enough. But I want to leave some safety margin for network glitches and occasional worker overload.

If your worker needs more than 1 second to echo back a string that was just typed even under ideal conditions, you really need to rethink the hardware setup.

Reply:
I've removed the timeout argument. Saving 85 seconds in a rare failure case is not worth this kind of discussion.